jlm/llvm/opt/alias-analyses/PointerObjectSet.hpp

```diff
@@ -44,15 +44,15 @@ class PointerObject final
 {
   PointerObjectKind Kind_ : util::BitWidthOfEnum(PointerObjectKind::COUNT);
 
-  // May point to a memory object defined outside of the module, or any escaped memory object
+  // May point to a memory object defined outside the module, or any escaped memory object
   uint8_t PointsToExternal_ : 1;
 
   // This memory object's address is known outside the module.
   // If set and Kind_ is Register, it only means the pointees of the Reigister have escaped.
   uint8_t HasEscaped_ : 1;
 
 public:
-  using Index = size_t;
+  using Index = std::uint32_t;
```

Comment:
Any particular reason why you changed that? The `size_t` guaranteed before that it is always the size of the register. Why was it important to know the exact size here? Considering also that you are actually hiding it behind `Index` again.

Reply:
This is due to the tendency of running out of memory while analyzing. Almost all our memory usage is sets containing `PointerObject::Index`, so we really save a lot of memory by doing this.

At the risk of making a "Bill Gates 640K statement", I think 4 billion variables + registers is going to be enough for any program.

All the points-to-sets need to be in memory while creating the final PointsToGraph, which becomes a bottleneck. It should also help speed, as the memory bandwidth hopefully decreases.